### ISSUE TITLE
MSI Installer with Auto Update

### ISSUE THREAD

**Issue by luisabtrace** [electron-userland/electron-builder#6965]
Hi,

This is more of a question/request for advice, so apologies for creating an issue.

I was wondering why doesn't electron-updater support auto update for MSI targets and what would it take for me to roll out this functionality on my own. What are the technical challenges preventing this feature? I don't understand anything about MSI, but conceptually it seems like auto update should work the same independently of the target: you find out if there is a new version, if there is one you download it and run it. Doesn't matter if it's a .msi or .exe. Perhaps I'm being clueless, but that's why I'm asking here.

Unfortunately some of our customers are asking for MSI installers, but I don't want to give up of auto update.

Thanks!

**Comment by mmaietta** [comment]
Hi @luisabtrace, I maintain electron-builder and honestly have no idea why support was never added. The project far predates my joining. I'm also not familiar with msi format to be able to chime on that portion either

**Comment by esimkowitz** [comment]
Necro, would love to see MSI added to electron-updater. Right now, if a user installs an app via MSI and then auto-updates, they'll have two separate installs on their system, which is really inconvenient

**Comment by esimkowitz** [comment]
Might have some time to investigate this in a month or so

**Comment by erseco** [comment]
Hi folks 👋

I'm currently planning to use the auto-update feature for my Electron app and noticed this open issue. Before going forward, I just wanted to ask if there's been any progress or insights on MSI support for auto updates.

Based on the current state, would you recommend sticking with NSIS for now to avoid issues with duplicate installations and broken update paths?

Thanks in advance for any updates or guidance!

**Comment by esimkowitz** [comment]
From my experience, I would recommend sticking with NSIS. There's no easy way to clean up the double installation issue without removing both of them and it'll just cause more confusion for your users. I've gotten pressure from some package managers to justify my usage of NSIS over MSI, just send them this issue for reference.

**Comment by pierremtb** [comment]
I believe this should ticket should remain open. Also interested in MSI with auto-updates.

**Comment by pierremtb** [comment]
Still believe this should remain open!

**Comment by mmaietta** [comment]
Closing as Not Planned. Unless someone else wants to pick this up, this is being closed.

**Comment by theIYD** [comment]
I have this case where an MSI needs to auto-update when a new version of the app (MSI only) is published to Github releases. 

The approach I have in mind is:
1. Write a script which runs in the background (CRON) and checks for the latest release on Github. 
2. If there is a new release, download it. 
3. Post download, force-quit the app with a message shown to the user for update is ready, silently install it using `msiexec` command and remove the existing version. 
4. Post installation, update the Windows registry with the new version of the app (I am not sure if this is needed)

Does this idea work? How do other apps do this? Zoom has an MSI version available online and I have seen that they have this Updater.exe.

**Comment by beyondkmp** [comment]
What do you think of Squirrel.Windows? This is an MSI that supports auto-update. Slack and VSCode use this for their auto-updating MSI.

**Comment by theIYD** [comment]
@beyondkmp How would Squirrel.Windows even help here? I need the app to be a machine-wide installation (app should be installed in C:\ProgramFiles) hence I need to build an MSI installer.

**Comment by mmaietta** [comment]
>I need the app to be a machine-wide installation

NSIS installers can do that and they support auto-update.
https://www.electron.build/nsis.html#permachine

**Comment by theIYD** [comment]
> > I need the app to be a machine-wide installation
> 
> NSIS installers can do that and they support auto-update. https://www.electron.build/nsis.html#permachine

True. But, an additional requirement I have is the app should be deployable in an enterprise setup i.e via ActiveDirectory and Group Policy. NSIS produce EXE which are not deployable.
